### PR TITLE
feat(PN-9571): adaptation of drag and drop component to WCAG 2.2 standards

### PR DESCRIPTION
**Pull request by SarahDonvito**
## Short description
Added CTA in _SingleFileInput_ component

### Preview
image

## List of changes proposed in this pull request
- Added button to component to make CTA more accessible
- Added prop for copy of button

## Product
Piattaforma Notifiche -> SEND

## How to test
Run storybook and go to SingleFileInput component to check that box has button to upload file

### CODE REVIEW

**Review by ZorbaDimatteo — CHANGES_REQUESTED** [review]
@SarahDonvito Error case scenario doesn't seem to be managed correctly
Screenshot 2024-04-09 at 11 19 31

File: src/components/SingleFileInput/SingleFileInput.tsx [hunk 1]
```diff
@@ -57,6 +58,9 @@ export type SingleFileInputProps = {
   /** The label to be displayed in the dropzone. */
   dropzoneLabel: string;
 
+  /** The button to be displayed in the dropzone. */
```
Comment (ZorbaDimatteo) [inline src/components/SingleFileInput/SingleFileInput.tsx:61]: @SarahDonvito Try to explain better this prop: is not the button per se (which is basically always included) but the label for the button

File: src/components/SingleFileInput/SingleFileInput.tsx [hunk 2]
```diff
@@ -233,6 +237,9 @@ export const SingleFileInput = ({
                   <Typography display="inline" variant="body2">
                     {dropzoneLabel}
                   </Typography>
+                  <Button sx={{ margin: "0 10px" }} variant="contained">
```
Comment (ZorbaDimatteo) [inline src/components/SingleFileInput/SingleFileInput.tsx:240]: @SarahDonvito this two block of code (STATUS.IDLE and STATUS.ERROR) are almost identical, except for `color` prop.
Try to improve code readability and reusability

**Review by ZorbaDimatteo — CHANGES_REQUESTED** [review]

**Review by ZorbaDimatteo — APPROVED** [review]

**Comment by ZorbaDimatteo** [comment]
@Carminepo2 @ruggerocastagnola Can you check this PR too?

**Comment by Carminepo2** [comment]
LGTM 👍

**Pull request merged**